core/elog/ali/writer.go

```diff
@@ -135,28 +128,50 @@ func (w *writer) write(fields map[string]interface{}) (err error) {
 }
 
 func (w *writer) flush() error {
-	// TODO sync pool
-	var lg = *w.group[0]
-	chlen := len(w.ch)
-	if chlen == 0 {
+	entriesChLen := len(w.ch)
+	if entriesChLen == 0 {
 		return nil
 	}
-	var logs = make([]*pb.Log, 0, chlen)
-	logs = append(logs, <-w.ch)
+
+	var waitedEntries = make([]*pb.Log, 0, entriesChLen)
+	waitedEntries = append(waitedEntries, <-w.ch)
 L1:
-	for {
+	for i := 0; i < entriesChLen; i++ {
 		select {
 		case l := <-w.ch:
-			logs = append(logs, l)
+			waitedEntries = append(waitedEntries, l)
 		default:
 			break L1
 		}
 	}
-	lg.Logs = logs
-	return w.store.PutLogs(&lg)
+
+	chunks := int(math.Ceil(float64(len(waitedEntries)) / float64(w.apiBulkSize)))
+	for i := 0; i < chunks; i++ {
+		go func(start int) {
+			end := (start + 1) * w.apiBulkSize
+			if end > len(waitedEntries) {
+				end = len(waitedEntries)
+			}
+			lg := pb.LogGroup{Logs: waitedEntries[start:end]}
+			if e := w.store.PutLogs(&lg); e != nil {
+				// if error occurs we put logs with fallbackCore logger
+				for _, v := range lg.Logs {
+					fields := make([]zapcore.Field, len(v.Contents), len(v.Contents))
+					for i, val := range v.Contents {
+						fields[i] = zap.String(val.GetKey(), val.GetValue())
+					}
+					if e := w.fallbackCore.Write(zapcore.Entry{Time: time.Now()}, fields); e != nil {
+						log.Println("fallbackCore write fail", e)
+					}
+				}
+			}
```

Comment:
could you please extract these lines to a new function?

Reply:
plase see <https://github.com/gotomicro/ego/pull/52/commits/d558e9eaa79f9a71b3d41f127de1734886d9c182#diff-e92af4b37a4557ab063f6cbe95d5a05fa0cd1d6facf9f040c6504955d531648aR156>.